torch/distributed/_tensor/placement_types.py

```diff
@@ -351,7 +351,7 @@ def __eq__(self, other: object) -> bool:
         return self.reduce_op == other.reduce_op
 
     def __hash__(self) -> int:
-        return hash(self.reduce_op)
+        return 1 + hash(self.reduce_op)
```

Comment:
oh, because if we don't have this one, Partial will have the the same hash as shard.

For Partial(SUM) has exactly same hash value as Shard(0). You can see the screenshot I listed in PR summary.

Reply:
I see, did you test `Shard(1)` and `Partial(SUM)` too and they are still different?